5.0 Draft/IDD/IDDs Service Registry/eu.arrowhead.service-discovery-http-json.yml

```diff
@@ -0,0 +1,631 @@
+        '414': { $ref: '#/components/responses/414-URITooLarge' }
+        '429': { $ref: '#/components/responses/429-TooManyRequests' }
+        '431': { $ref: '#/components/responses/431-RequestHeaderFieldsTooLarge' }
+        '500': { $ref: '#/components/responses/500-InternalServerError' }
+        '503': { $ref: '#/components/responses/503-ServiceUnavailable' }
+
+components:
+
+  headers:
+
+    Cache-Control:
+      description: |
+        If present, the value of this header may only include the `max-age`, `must-revalidate` and
+        `no-store` response directives, all of which are defined in
+        [RFC 9111 Section 5.2.2](https://datatracker.ietf.org/doc/html/rfc9111#name-response-directives).
+
+        The `max-age` directive indicates for how many seconds the records in the response should be
+        cached before being queried again by the consumer. The `must-revalidate` directive is used
+        to forbid the consumer from using the records after the `max-age` has expired and the
+        attempt to fetch them again failed. The `no-store` directive instructs the consumer not to
+        cache the records at all.
+      schema:
+        type: string
+      example: max-age=1800, must-revalidate
+      required: false
+
+    Limit:
+      description: |
+        An indication of the maximum number of records the consuming system is allowed to have
+        included in the response to its query.
+      schema:
+        type: number
+        format: int32
+        minimum: 0
+      required: true
+
+    Retry-After:
+      description: |
+        An indication, in seconds, of how long a consumer is to wait before attempting the failed
+        request again.
+      schema:
+        type: number
+        format: int32
+        minimum: 1
+      required: true
+
+  parameters:
+
+    deregister-q:
+      name: q
+      in: query
+      description: |
+        Deregistration query, consisting of an array of comma-separated deregistration expressions.
+        Every record matching any of the individual expressions will be deregistered, if the
+        consumer has the appropriate permissions.
+      example: eu.arrowhead.service-discovery-http-json,eu.arrowhead.authorization-http-json@MWtzOXIpaUEsOyEhc1NsTwo=
+      schema:
+        type: array
+        items: { $ref: '#/components/schemas/DeregisterExpression' }
+      explode: false
+      allowReserved: true
+
+    query-limit:
+      name: limit
+      in: query
+      description: |
+        An indication of the maximum number of records a consuming system wants to have included in
+        the response to its query.
+      example: 24
+      schema:
+        description: |
+          Indicates a maximum number of records a consuming service wishes to have included in the
+          response to some query.
+        type: number
+        format: int32
+        minimum: 1
+
+    query-offset:
+      name: offset
```

Comment:
We could consider using cursor-based pagination instead since it is more scalable and puts less constraints on the implementing solution.